src/test/miner_tests.cpp

```diff
@@ -270,29 +270,37 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
     tx.vin[0].prevout.n = 0;
     tx.vout.resize(1);
     tx.vout[0].nValue = BLOCKSUBSIDY;
+
+    // Creates 1001 multisig transactions
+    std::vector<CTransaction> multisigTxs;
+
     for (unsigned int i = 0; i < 1001; ++i)
     {
         tx.vout[0].nValue -= LOWFEE;
-        hash = tx.GetHash();
+        CTransaction cTx(tx);
+        multisigTxs.push_back(cTx);
+        tx.vin[0].prevout.hash = cTx.GetHash();
+    }
+
+    for (unsigned int i = 0; i < multisigTxs.size(); ++i)
+    {
+        CTransaction &cTx = multisigTxs[i];
+        hash = cTx.GetHash();
         bool spendsCoinbase = i == 0; // only first tx spends coinbase
         // If we don't set the # of sig ops in the CTxMemPoolEntry, template creation fails
-        mempool.addUnchecked(hash, entry.Fee(LOWFEE).Time(GetTime()).SpendsCoinbase(spendsCoinbase).FromTx(tx));
-        tx.vin[0].prevout.hash = hash;
+        mempool.addUnchecked(hash, entry.Fee(LOWFEE).Time(GetTime()).SpendsCoinbase(spendsCoinbase).FromTx(cTx));
```

Comment:
Actually in the original code it doesn't since each `CMutableTransaction` is converted to a `CTransaction` just once, and then the hash is cached and copying the `CTransaction` just copies the cached hash. Removing the method that accepts `CTransaction` makes that optimization impossible.
Anyway, since the method is no longer there, I'll remove this commit and think of some other way of speeding up things in a separate PR.

Reply:
Yes, you are right. Feel free to reintroduce the method if you think it was helpful. However, I think if you keep a vector of CTransactionRef as `multisig_txs` instead your patch should work just fine?